### PR TITLE
docs: Add documentation for deployment projects

### PR DESCRIPTION
**Pull request by normj**
*Description of changes:*
Rewrote the docs for deployment projects going into depth on the recipe file, how to customize the CDK project and providing a tutorial for creating a deployment project and making modifications.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by philasmar — APPROVED** [review]

**Comment by ashovlin** [comment]
I've added a second commit to this branch (and force-pushed) after an internal discussion with @ik130 , to try to tweak this with more an "outsider" perspective since I didn't work on the original implementation of this feature.

Notable changes:
* Moved from "deployment project" to "custom deployment project" in many occurrences, to try to emphasize the difference between the application being deployed and the built-in recipes
* Moved the tutorial up in the TOC before the larger, reference-style pages
* Some typo fixes and other small tweaks

File: site/content/docs/deployment-projects/cdk-project.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+The following links are useful resources to learn how to write .NET CDK projects.
+
+* [.NET CDK Workshop](https://cdkworkshop.com/40-dotnet.html)
+* [.NET CDK Reference](https://docs.aws.amazon.com/cdk/api/v2/dotnet/api/index.html)
+* [AWS CDK Developer Guide](https://docs.aws.amazon.com/cdk/v2/guide/home.html)
+
+The main concepts to understand about the CDK are the following.
+
+* Stack - this is the top level container for all of the AWS resources represented as constructs. In deployment projects this type is usually the `AppStack` type in the project.
+* Constructs - AWS resources are modeled as constructs. Some constructs are a one-to-one mapping to a single AWS resource. The CDK also has higher level abstractions where a single construct represents a collection of AWS resources that solve a specific problem.
```
Comment (normj) [inline site/content/docs/deployment-projects/cdk-project.md:16]: nit: Replace "specific" with "common". Makes it sounds more useful

File: site/content/docs/deployment-projects/cdk-project.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* Construct properties - For every construct this is a properties object that is created first, all necessary values are set and then passed into the constructor of a construct.
```
Comment (normj) [inline site/content/docs/deployment-projects/cdk-project.md:17]: `this is` -> `there is`

**Review by normj — COMMENTED** [review]
I can't click the approve button since I'm the original create of the PR but I approve the PR with 2 small text change suggestions.

File: site/content/docs/deployment-projects/cdk-project.md (hunk 1 above)
Comment (ashovlin) [reply site/content/docs/deployment-projects/cdk-project.md]: Addressed in new commit.

File: site/content/docs/deployment-projects/cdk-project.md (hunk 2 above)
Comment (ashovlin) [reply site/content/docs/deployment-projects/cdk-project.md]: Addressed in new commit.

**Review by ashovlin — APPROVED** [review]

**Review by ik130 — APPROVED** [review]

**Pull request merged**